### ISSUE TITLE
Too easy to apply edits to all taks (implicit status:pending for display, but not for edits)

### ISSUE THREAD

**Issue by jodavaho** [GothenburgBitFactory/taskwarrior#2112]
Try this:  `task proj:home`

I can now view all tasks in my home project. (note, it displays those who match `status:pending` as though I had put that in). 

Suppose after viewing these tasks, I want to be ambitious and finish them all before the end of the week.

`task proj:home modify due:eow`

Now, it changes all my past tasks regardless of the implicit `status:pending` in the first query. Note, it might be something technically other than `status:pending`, but I hope the example is clear and I want to continue as though that were the case.

Problems:

- Viewing the set of tasks before modifying is probably a good practice. It is obfuscated that we should including `status:pending` in one but no the other. I'd argue the usage *encourages* this type of error.
- Worse, it seems to prompt me about tasks matching `status:pending` first, giving a false sense of security that selecting the `all` option is OK, which allows the edit to proceed on all past tasks as before.

Impact:

- Analysis tools that look at past completion / on-time rates *require* knowledge of due dates, scheduled dates, and in general *every other attribute* to remain unchanged after closing. One slip, and I've overwritten all that.

Request:

- Change status:pending (or equivalent) to be implicit consistently across all commands. Inform users that status:closed should be included to edit past tasks

**Comment by janikrabe** [comment]
I agree that this is an issue; taskwarrior does not follow the principle of least astonishment here. I solved this using `alias tp='task +PENDING'`, but I still have to remember to use `tp` every time I want to modify only pending tasks.

I'm not sure how easy implementing `status:closed` would be since `closed` currently isn't a valid status (as opposed to `completed` and `deleted`). Additionally, `status:closed` sounds like it would *only* modify closed tasks, while a user may also want to modify pending and waiting tasks.

Another option would be to add an `--all` flag, but since taskwarrior doesn't currently have many (or any?) flags, I'm not sure that's the best solution.

**Comment by pbeckingham** [comment]
The example is misleading because you are relying on unstated defaults. It is more accurate to compare these two commands:

    $ task proj:home $(task _get rc.report.$(task _get rc.default.command).filter)
    $ task proj:home modify due:eow

This makes it more clear that it should be expected that the two commands reference different subsets. Simply added `status:pending` is only right for a default command happens to have that filter.

So if the `modify` command is expected to obey the filter of the default command, how is that overridden?

This is problematic.
